suite/tests/client-interface/drx-scattergather.c

```diff
@@ -176,7 +196,19 @@ test_avx2_gather(void (*test_func)(uint32_t *, uint32_t *, uint32_t *),
                  uint32_t *test_idx_vec, uint32_t *output_xmm_ymm OUT)
 {
     memset(output_xmm_ymm, 0, CONCAT_XMM_YMM_U32 * sizeof(uint32_t));
+#    ifdef UNIX
+    byte ymm_buf[32];
+    memset(ymm_buf, 0, sizeof(ymm_buf));
+    byte ref_buf[32];
+    memset(ref_buf, 0, sizeof(ref_buf));
     test_func(ref_sparse_test_buf, test_idx_vec, output_xmm_ymm);
+    /* XXX i#2985: add check to non-UNIX systems. */
+    __asm__ __volatile__("vmovdqu %%ymm2, %0" : : "m"(ymm_buf) : "ymm2");
+    if (memcmp(ymm_buf, ref_buf, sizeof(ymm_buf)) != 0) {
```

Comment:
Another one that I find surprising given the name "ref_buf": if hardcoding a comparison to 0, maybe s/ref_buf/zero_buf/.

Reply:
Done.